### PR TITLE
Add aws_eks_cluster resource

### PR DESCRIPTION
**Pull request by clintoncwolfe**
This PR is an adoption of #3504, originally authored by @j00p34. That PR had reached approval by me, with the exception of the integration tests.

My modifications were just around the Terraform code used in the integration tests, some adjustments of the integration tests, and a modification to exception handling.

The code now passes unit and integration tests.

screen shot 2018-11-06 at 18 27 30

### CODE REVIEW

File: lib/resources/aws/aws_eks_cluster.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  example <<-EOX
+    describe aws_eks_cluster('default') do
+      it { should exist }
+    end
+EOX
+  supports platform: 'aws'
+
+  include AwsSingularResourceMixin
+  attr_reader :version, :arn, :cluster_name, :certificate_authority, :name,
```
Comment (jquick) [inline lib/resources/aws/aws_eks_cluster.rb:13]: Would prefer alphabetized here. Makes it easier to find things later.

**Review by jquick — APPROVED** [review]
Looks good, just one suggestion. Thanks!

**Review by jerryaldrichiii — APPROVED** [review]
Great work @clintoncwolfe!

**Comment by jerryaldrichiii** [comment]
Also, thank you @j00p34 for getting this started!

**Pull request merged**